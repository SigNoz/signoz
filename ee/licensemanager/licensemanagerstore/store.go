package licensemanagerstore

import (
	"context"

	"github.com/SigNoz/signoz/ee/types/licensetypes"
	"github.com/SigNoz/signoz/pkg/errors"
	"github.com/SigNoz/signoz/pkg/sqlstore"
	"github.com/SigNoz/signoz/pkg/types"
	"github.com/SigNoz/signoz/pkg/types/featuretypes"
	"github.com/SigNoz/signoz/pkg/valuer"
)

type store struct {
	sqlstore sqlstore.SQLStore
}

func New(sqlstore sqlstore.SQLStore) licensetypes.Store {
	return &store{sqlstore}
}

func (s *store) Create(ctx context.Context, storableLicense *licensetypes.StorableLicense) error {
	_, err := s.
		sqlstore.
		BunDB().
		NewInsert().
		Model(storableLicense).
		Exec(ctx)
	if err != nil {
		return s.sqlstore.WrapAlreadyExistsErrf(err, errors.CodeInternal, "unable to create the license with ID: %s", storableLicense.ID)
	}

	return nil
}

func (s *store) Get(ctx context.Context, organizationID valuer.UUID, licenseID valuer.UUID) (*licensetypes.StorableLicense, error) {
	storableLicense := new(licensetypes.StorableLicense)
	err := s.
		sqlstore.
		BunDB().
		NewSelect().
		Model(storableLicense).
		Where("org_id = ?", organizationID).
		Where("id = ?", licenseID).
		Scan(ctx)
	if err != nil {
		return nil, s.sqlstore.WrapNotFoundErrf(err, errors.CodeInternal, "unable to fetch the license with ID: %s", licenseID)
	}

	return storableLicense, nil
}

func (s *store) GetAll(ctx context.Context, organizationID valuer.UUID) ([]*licensetypes.StorableLicense, error) {
	storableLicenses := make([]*licensetypes.StorableLicense, 0)
	err := s.
		sqlstore.
		BunDB().
		NewSelect().
		Model(&storableLicenses).
		Where("org_id = ?", organizationID).
		Scan(ctx)
	if err != nil {
		return nil, s.sqlstore.WrapNotFoundErrf(err, errors.CodeInternal, "unable to fetch the licenses for organization with ID: %s", organizationID)
	}

	return storableLicenses, nil
}

func (s *store) Update(ctx context.Context, storableLicense *licensetypes.StorableLicense) error {
	_, err := s.
		sqlstore.
		BunDB().
		NewUpdate().
		Model(storableLicense).
		Column("data", "last_validated_at").
		WherePK().
		Exec(ctx)
	if err != nil {
		return errors.Wrapf(err, errors.TypeInternal, errors.CodeInternal, "unable to update license with ID: %s", storableLicense.ID)
	}

	return nil
}

func (s *store) ListOrganizations(ctx context.Context) ([]valuer.UUID, error) {
	orgIDStrs := make([]string, 0)
	err := s.sqlstore.
		BunDB().
		NewSelect().
		Model(new(types.Organization)).
		Column("id").
		Scan(ctx, &orgIDStrs)
	if err != nil {
		return nil, err
	}

	orgIDs := make([]valuer.UUID, len(orgIDStrs))
	for idx, orgIDStr := range orgIDStrs {
		orgID, err := valuer.NewUUID(orgIDStr)
		if err != nil {
			return nil, err
		}
		orgIDs[idx] = orgID
	}

	return orgIDs, nil

}

func (s *store) CreateFeature(ctx context.Context, storableFeature *featuretypes.StorableFeature) error {
	_, err := s.
		sqlstore.
		BunDB().
		NewInsert().
		Model(storableFeature).
		Exec(ctx)
	if err != nil {
		return s.sqlstore.WrapAlreadyExistsErrf(err, errors.CodeInternal, "unable to create feature with name: %s", storableFeature.Name)
	}

	return nil
}

func (s *store) GetFeature(ctx context.Context, key string) (*featuretypes.StorableFeature, error) {
	storableFeature := new(featuretypes.StorableFeature)
	err := s.
		sqlstore.
		BunDB().
		NewSelect().
		Model(storableFeature).
		Where("name = ?", key).
		Scan(ctx)
	if err != nil {
		return nil, s.sqlstore.WrapNotFoundErrf(err, errors.CodeInternal, "unable to fetch the feature with key: %s", key)
	}

	return storableFeature, nil
}

func (s *store) GetAllFeatures(ctx context.Context) ([]*featuretypes.StorableFeature, error) {
	storableFeatures := make([]*featuretypes.StorableFeature, 0)
	err := s.
		sqlstore.
		BunDB().
		NewSelect().
		Model(&storableFeatures).
		Scan(ctx)
	if err != nil {
		return nil, s.sqlstore.WrapNotFoundErrf(err, errors.CodeInternal, "unable to fetch all features")
	}

	return storableFeatures, nil
}

func (s *store) InitFeatures(ctx context.Context, storableFeatures []*featuretypes.StorableFeature) error {
	_, err := s.
		sqlstore.
		BunDB().
		NewInsert().
		Model(&storableFeatures).
		Exec(ctx)
	if err != nil {
		return s.sqlstore.WrapAlreadyExistsErrf(err, errors.CodeInternal, "unable to init features")
	}

	return nil
}

func (s *store) UpdateFeature(ctx context.Context, storableFeature *featuretypes.StorableFeature) error {
	_, err := s.
		sqlstore.
		BunDB().
		NewUpdate().
		Model(storableFeature).
		Exec(ctx)
	if err != nil {
		return errors.Wrapf(err, errors.TypeInternal, errors.CodeInternal, "unable to update feature with key: %s", storableFeature.Name)
	}

	return nil
}
