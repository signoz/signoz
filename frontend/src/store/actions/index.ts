export * from './types';
export * from './traceFilters';
export * from './serviceMap';
export * from './traces';
export * from './MetricsActions';
export * from './usage';
export * from './global';
