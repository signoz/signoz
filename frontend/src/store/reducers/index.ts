import { combineReducers } from 'redux';
import {
	traceResponseNew,
	spansWSameTraceIDResponse,
	usageDataItem,
	GlobalTime,
	serviceMapStore,
	TraceFilters,
} from '../actions';
import { updateGlobalTimeReducer } from './global';
import { MetricsInitialState, metricsReducer } from './metrics';
import TraceFilterReducer from './traceFilters';
import { traceItemReducer, tracesReducer } from './traces';
import { usageDataReducer } from './usage';
import { ServiceMapReducer } from './serviceMap';

export interface StoreState {
	metricsData: MetricsInitialState;
	traceFilters: TraceFilters;
	traces: traceResponseNew;
	traceItem: spansWSameTraceIDResponse;
	usageDate: usageDataItem[];
	globalTime: GlobalTime;
	serviceMap: serviceMapStore;
}

const reducers = combineReducers<StoreState>({
	traceFilters: TraceFilterReducer,
	traces: tracesReducer,
	traceItem: traceItemReducer,
	usageDate: usageDataReducer,
	globalTime: updateGlobalTimeReducer,
	metricsData: metricsReducer,
	serviceMap: ServiceMapReducer,
});

export default reducers;
