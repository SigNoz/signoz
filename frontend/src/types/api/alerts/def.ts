import { AlertLabelsProps } from 'pages/AlertDetails/AlertHeader/AlertLabels/AlertLabels';
import { ICompositeMetricQuery } from 'types/api/alerts/compositeQuery';

// default match type for threshold
export const defaultMatchType = '1';

// default eval window
export const defaultEvalWindow = '5m0s';

export const defaultFrequency = '1m0s';

// default compare op: above
export const defaultCompareOp = '1';

export const defaultAlgorithm = 'standard';

export const defaultSeasonality = 'hourly';

export interface AlertDef {
	id?: number;
	alertType?: string;
	alert: string;
	ruleType?: string;
	frequency?: string;
	condition: RuleCondition;
	labels?: Labels;
	annotations?: Labels;
	evalWindow?: string;
	source?: string;
	disabled?: boolean;
	preferredChannels?: string[];
	broadcastToAll?: boolean;
	version?: string;
}

export interface RuleCondition {
	compositeQuery: ICompositeMetricQuery;
	op?: string;
	target?: number;
	matchType?: string;
	targetUnit?: string;
	selectedQueryName?: string;
	alertOnAbsent?: boolean | undefined;
	absentFor?: number | undefined;
	requireMinPoints?: boolean | undefined;
	requiredNumPoints?: number | undefined;
	algorithm?: string;
	seasonality?: string;
}
export interface Labels {
	[key: string]: string;
}

export interface AlertRuleStats {
	totalCurrentTriggers: number;
	totalPastTriggers: number;
	currentTriggersSeries: CurrentTriggersSeries;
	pastTriggersSeries: CurrentTriggersSeries | null;
	currentAvgResolutionTime: number;
	pastAvgResolutionTime: number;
	currentAvgResolutionTimeSeries: CurrentTriggersSeries;
	pastAvgResolutionTimeSeries: any | null;
}

interface CurrentTriggersSeries {
	labels: Labels;
	labelsArray: any | null;
	values: StatsTimeSeriesItem[];
}

export interface StatsTimeSeriesItem {
	timestamp: number;
	value: string;
}

export type AlertRuleStatsPayload = {
	data: AlertRuleStats;
};

export interface AlertRuleTopContributors {
	fingerprint: number;
	labels: Labels;
	count: number;
	relatedLogsLink: string;
	relatedTracesLink: string;
}
export type AlertRuleTopContributorsPayload = {
	data: AlertRuleTopContributors[];
};

export interface AlertRuleTimelineTableResponse {
	ruleID: string;
	ruleName: string;
	overallState: string;
	overallStateChanged: boolean;
	state: string;
	stateChanged: boolean;
	unixMilli: number;
	labels: Labels;
	fingerprint: number;
	value: number;
	relatedTracesLink: string;
	relatedLogsLink: string;
}
export type AlertRuleTimelineTableResponsePayload = {
	data: {
		items: AlertRuleTimelineTableResponse[];
		total: number;
		labels: AlertLabelsProps['labels'];
	};
};
type AlertState = 'firing' | 'normal' | 'no-data' | 'muted';

export interface AlertRuleTimelineGraphResponse {
	start: number;
	end: number;
	state: AlertState;
}
export type AlertRuleTimelineGraphResponsePayload = {
	data: AlertRuleTimelineGraphResponse[];
};
