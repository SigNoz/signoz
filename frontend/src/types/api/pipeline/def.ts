import { TagFilter } from '../queryBuilder/queryBuilderData';

export interface ProcessorData {
	type: string;
	id?: string;
	orderId: number;
	name: string;
	enabled?: boolean;
	output?: string;
	parse_to?: string;
	pattern?: string;
	parse_from?: string;
	from?: string;
	to?: string;
	regex?: string;
	on_error?: string;
	field?: string;
	value?: string;

	// trace parser fields.
	trace_id?: {
		parse_from: string;
	};
	span_id?: {
		parse_from: string;
	};
	trace_flags?: {
		parse_from: string;
	};

	// time parser fields
	layout_type?: string;
	layout?: string;

	// json flattening fields
	enable_flattening?: boolean;
	enable_paths?: boolean;
	path_prefix?: string;
	enable_mapping?: boolean;
	mapping?: Record<string, string[]>;
}

export interface PipelineData {
	alias: string;
	config?: Array<ProcessorData>;
	createdAt: string;
	description?: string;
	createdBy: string;
	enabled: boolean;
	filter: TagFilter;
	id?: string;
	name: string;
	orderId: number;
	tags?: Array<string>; // Tags data is missing in API response
}

export interface HistoryData {
	active: boolean;
	createdAt: string;
	createdBy: string;
	createdByName: string;
	deployStatus: string;
	deployResult: string;
	disabled: boolean;
	elementType: string;
	id: string;
	isValid: boolean;
	lastConf: string;
	lastHash: string;
	version: number;
}

export interface Pipeline {
	active: boolean;
	createdBy: string;
	deployResult: string;
	deployStatus: string;
	disabled: boolean;
	elementType: string;
	history: Array<HistoryData>;
	id: string;
	is_valid: boolean;
	lastConf: string;
	lastHash: string;
	pipelines: Array<PipelineData>;
	version: string | number;
}

export enum ActionType {
	AddPipeline = 'add-pipeline',
	EditPipeline = 'edit-pipeline',
	AddProcessor = 'add-processor',
	EditProcessor = 'edit-processor',
}

export enum ActionMode {
	Viewing = 'viewing-mode',
	Editing = 'editing-mode',
	Deploying = 'deploying-mode',
}
