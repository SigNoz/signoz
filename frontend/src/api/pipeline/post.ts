import axios from 'api';
import { ErrorResponseHandlerV2 } from 'api/ErrorResponseHandlerV2';
import { AxiosError } from 'axios';
import { ErrorV2Resp, SuccessResponseV2 } from 'types/api';
import { Pipeline } from 'types/api/pipeline/def';
import { Props } from 'types/api/pipeline/post';

const post = async (props: Props): Promise<SuccessResponseV2<Pipeline>> => {
	try {
		const response = await axios.post('/logs/pipelines', props.data);

		return {
			httpStatusCode: response.status,
			data: response.data.data,
		};
	} catch (error) {
		ErrorResponseHandlerV2(error as AxiosError<ErrorV2Resp>);
	}
};

export default post;
