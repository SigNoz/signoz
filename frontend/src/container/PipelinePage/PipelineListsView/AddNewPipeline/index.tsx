import { Button, Divider, Form, Modal } from 'antd';
import { useAppContext } from 'providers/App/App';
import React, { useCallback, useEffect, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { ActionMode, ActionType, PipelineData } from 'types/api/pipeline/def';
import { v4 } from 'uuid';

import { ModalButtonWrapper, ModalTitle } from '../styles';
import { getEditedDataSource, getRecordIndex } from '../utils';
import { renderPipelineForm } from './utils';

function AddNewPipeline({
	isActionType,
	setActionType,
	selectedPipelineData,
	setShowSaveButton,
	setCurrPipelineData,
	currPipelineData,
}: AddNewPipelineProps): JSX.Element {
	const [form] = Form.useForm();
	const { t } = useTranslation('pipeline');
	const { user } = useAppContext();

	const isEdit = isActionType === 'edit-pipeline';
	const isAdd = isActionType === 'add-pipeline';

	useEffect(() => {
		if (isEdit) {
			form.setFieldsValue(selectedPipelineData);
		}
		if (isAdd) {
			form.resetFields();
		}
	}, [form, isEdit, isAdd, selectedPipelineData]);

	const onFinish = (values: PipelineData): void => {
		const newPipeLineData: PipelineData = {
			id: v4(),
			orderId: (currPipelineData?.length || 0) + 1,
			createdAt: new Date().toISOString(),
			createdBy: user?.name || '',
			name: values.name,
			alias: values.name.replace(/\s/g, ''),
			description: values.description,
			filter: values.filter,
			config: [],
			enabled: true,
		};

		if (isEdit && selectedPipelineData) {
			const findRecordIndex = getRecordIndex(
				currPipelineData,
				selectedPipelineData,
				'id',
			);
			const updatedPipelineData: PipelineData = {
				...currPipelineData[findRecordIndex],
				...values,
			};

			const editedPipelineData = getEditedDataSource(
				currPipelineData,
				selectedPipelineData,
				'id',
				updatedPipelineData,
			);

			setCurrPipelineData(editedPipelineData);
		}
		if (isAdd) {
			setCurrPipelineData((prevState) => {
				if (prevState) return [...prevState, newPipeLineData];
				return [newPipeLineData];
			});
		}
		setActionType(undefined);
	};

	const onCancelModalHandler = (): void => {
		setActionType(undefined);
	};

	const modalTitle = useMemo(
		(): string =>
			isEdit
				? `${t('edit_pipeline')} : ${selectedPipelineData?.name}`
				: t('create_pipeline'),
		[isEdit, selectedPipelineData?.name, t],
	);

	const onOkModalHandler = useCallback(
		() => setShowSaveButton(ActionMode.Editing),
		[setShowSaveButton],
	);

	const isOpen = useMemo(() => isEdit || isAdd, [isAdd, isEdit]);

	return (
		<Modal
			title={<ModalTitle level={4}>{modalTitle}</ModalTitle>}
			centered
			open={isOpen}
			width={800}
			footer={null}
			onCancel={onCancelModalHandler}
		>
			<Divider plain />
			<Form
				name="add-new-pipeline"
				layout="vertical"
				onFinish={onFinish}
				autoComplete="off"
				form={form}
			>
				{renderPipelineForm()}
				<Divider plain />
				<Form.Item>
					<ModalButtonWrapper>
						<Button
							key="submit"
							type="primary"
							htmlType="submit"
							onClick={onOkModalHandler}
						>
							{isEdit ? t('update') : t('create')}
						</Button>
						<Button key="cancel" onClick={onCancelModalHandler}>
							{t('cancel')}
						</Button>
					</ModalButtonWrapper>
				</Form.Item>
			</Form>
		</Modal>
	);
}

interface AddNewPipelineProps {
	isActionType: string;
	setActionType: (actionType?: ActionType) => void;
	selectedPipelineData: PipelineData | undefined;
	setShowSaveButton: (actionMode: ActionMode) => void;
	setCurrPipelineData: (
		value: React.SetStateAction<Array<PipelineData>>,
	) => void;
	currPipelineData: Array<PipelineData>;
}

export default AddNewPipeline;
