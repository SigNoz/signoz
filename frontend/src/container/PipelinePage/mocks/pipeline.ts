import { Pipeline, PipelineData } from 'types/api/pipeline/def';
import { DataTypes } from 'types/api/queryBuilder/queryAutocompleteResponse';
import { TagFilter } from 'types/api/queryBuilder/queryBuilderData';

export const configurationVersion = '1.0';

export function mockPipelineFilter(
	key: string,
	op: string,
	value: string,
): TagFilter {
	return {
		op: 'AND',
		items: [
			{
				id: `${key}-${value}`,
				key: {
					key,
					dataType: DataTypes.String,
					type: '',
					isColumn: false,
					isJSON: false,
				},
				op,
				value,
			},
		],
	};
}

export const pipelineMockData: Array<PipelineData> = [
	{
		id: '4453c8b0-c0fd-42bf-bf09-7cc1b04ccdc9',
		orderId: 1,
		name: 'Apache common parser',
		alias: 'apachecommonparser',
		description: 'This is a desc',
		enabled: false,
		filter: mockPipelineFilter('source', '=', 'nginx'),
		config: [
			{
				orderId: 1,
				enabled: true,
				type: 'grok_parser',
				id: 'grokusecommon',
				name: 'grok use common asd',
				output: 'renameauth',
				parse_to: 'attributes',
				pattern: '%{COMMONAPACHELOG}',
				parse_from: 'body',
			},
			{
				orderId: 2,
				enabled: true,
				type: 'move',
				id: 'renameauth',
				name: 'rename auth',
				from: 'attributes.auth',
				to: 'attributes.username',
			},
			{
				orderId: 3,
				enabled: true,
				type: 'json_parser',
				id: 'jsonparser',
				name: 'json parser',
				from: 'attributes.auth',
				to: 'attributes.username',
			},
		],
		createdBy: 'nityananda@signoz.io',
		createdAt: '2023-03-07T16:56:53.36071141Z',
	},
	{
		id: 'a3675a0c-ff73-4ddb-be39-4351ace69231',
		orderId: 2,
		name: 'Moving pipeline new',
		alias: 'movingpipelinenew',
		description: 'This is a desc of move',
		enabled: false,
		filter: mockPipelineFilter('method', '=', 'POST'),
		config: [
			{
				orderId: 1,
				enabled: true,
				type: 'copy',
				id: 'mv1',
				name: 'mymove',
				from: 'attributes.method',
				to: 'attributes.method11',
			},
		],
		createdBy: 'chintan@signoz.io',
		createdAt: '2023-03-07T16:55:27.789595116Z',
	},
];

export const pipelineApiResponseMockData: Pipeline = {
	id: '67ace08a-6b6c-4221-ab58-a5d3bd5eb6f2',
	version: 5,
	elementType: 'log_pipelines',
	createdBy: 'test@signoz.io',
	active: false,
	is_valid: false,
	disabled: false,
	deployStatus: 'IN_PROGRESS',
	deployResult: 'Deployment started',
	lastHash: 'q��҂�&覣ʝup��\u0003���q�6�\u001e�<ѥIb�',
	lastConf:
		'[{"id":"a3675a0c-ff73-4ddb-be39-4351ace69231","orderId":"1","name":"Moving pipeline new","alias":"movingpipelinenew","enabled":false,"filter":"attributes.method == POST","config":[{"type":"copy","id":"mv1","name":"mymove","from":"attributes.method","to":"attributes.method11"}],"createdBy":"nityananda@signoz.io","createdAt":"2023-03-07T16:55:27.789595116Z"},{"id":"4453c8b0-c0fd-42bf-bf09-7cc1b04ccdc9","orderId":"2","name":"Apache common parser","alias":"apachecommonparser","enabled":false,"filter":"attributes.source == nginx","config":[{"type":"grok_parser","id":"grokusecommon","name":"grok use common asd","output":"renameauth","parse_to":"attributes","pattern":"%{COMMONAPACHELOG}","parse_from":"body"},{"type":"move","id":"renameauth","name":"rename auth","from":"attributes.auth","to":"attributes.username"}],"createdBy":"nityananda@signoz.io","createdAt":"2023-03-07T16:56:53.36071141Z"}]',
	pipelines: pipelineMockData,
	history: [
		{
			id: 'e118dedd-e996-455a-9cb2-5bf50b77fc35',
			version: 1,
			elementType: 'log_pipelines',
			createdBy: 'test2@signoz.io',
			active: false,
			isValid: false,
			disabled: false,
			deployStatus: 'DEPLOYED',
			deployResult: 'deploy successful',
			lastHash: '',
			lastConf: '',
			// eslint-disable-next-line sonarjs/no-duplicate-string
			createdAt: '2021-03-07T16:56:53.36071141Z',
			createdByName: 'test2',
		},
		{
			id: '9a98673d-b5db-4281-89d3-d85ed9ffe311',
			version: 2,
			elementType: 'log_pipelines',
			createdBy: 'test3@signoz.io',
			active: false,
			isValid: false,
			disabled: false,
			deployStatus: 'DEPLOYED',
			deployResult: 'deploy successful',
			lastHash: '',
			lastConf: '',
			createdAt: '2021-03-07T16:56:53.36071141Z',
			createdByName: 'test3',
		},
		{
			id: '9fdb0813-f77f-4837-815e-bb6eedd64f68',
			version: 3,
			elementType: 'log_pipelines',
			createdBy: 'nityananda+1@signoz.io',
			active: false,
			isValid: false,
			disabled: false,
			deployStatus: 'IN_PROGRESS',
			deployResult: '',
			lastHash: '',
			lastConf: '',
			createdAt: '2021-03-07T16:56:53.36071141Z',
			createdByName: 'nityananda+1',
		},
		{
			id: '87efb1cf-85b0-4aa4-934e-62a118fa4ec7',
			version: 4,
			elementType: 'log_pipelines',
			createdBy: 'nityananda+2@signoz.io',
			active: false,
			isValid: false,
			disabled: false,
			deployStatus: 'IN_PROGRESS',
			deployResult: '',
			lastHash: '',
			lastConf: '',
			createdAt: '2021-03-07T16:56:53.36071141Z',
			createdByName: 'nityananda+2',
		},
		{
			id: '67ace08a-6b6c-4221-ab58-a5d3bd5eb6f2',
			version: 5,
			elementType: 'log_pipelines',
			createdBy: 'nityananda+4@signoz.io',
			active: false,
			isValid: false,
			disabled: false,
			deployStatus: 'IN_PROGRESS',
			deployResult: '',
			lastHash: '',
			lastConf: '',
			createdAt: '2021-03-07T16:56:53.36071141Z',
			createdByName: 'nityananda+4',
		},
	],
};
