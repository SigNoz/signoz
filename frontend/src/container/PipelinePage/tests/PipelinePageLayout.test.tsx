import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';
import { Pipeline } from 'types/api/pipeline/def';
import { v4 } from 'uuid';

import PipelinePageLayout from '../Layouts/Pipeline';
import { matchMedia } from './AddNewPipeline.test';

jest.mock('uplot', () => {
	const paths = {
		spline: jest.fn(),
		bars: jest.fn(),
	};
	const uplotMock = jest.fn(() => ({
		paths,
	}));
	return {
		paths,
		default: uplotMock,
	};
});

beforeAll(() => {
	matchMedia();
});

const queryClient = new QueryClient({
	defaultOptions: {
		queries: {
			refetchOnWindowFocus: false,
		},
	},
});

describe('PipelinePage container test', () => {
	it('should render PipelinePageLayout section', () => {
		const pipelinedata: Pipeline = {
			active: true,
			createdBy: 'admin',
			deployResult: 'random_data',
			deployStatus: 'random_data',
			disabled: false,
			elementType: 'random_data',
			history: [],
			id: v4(),
			is_valid: true,
			lastConf: 'random_data',
			lastHash: 'random_data',
			pipelines: [],
			version: 1,
		};

		const refetchPipelineLists = jest.fn();

		const { asFragment } = render(
			<MemoryRouter>
				<QueryClientProvider client={queryClient}>
					<Provider store={store}>
						<I18nextProvider i18n={i18n}>
							<PipelinePageLayout
								pipelineData={pipelinedata}
								refetchPipelineLists={refetchPipelineLists}
							/>
						</I18nextProvider>
					</Provider>
				</QueryClientProvider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
