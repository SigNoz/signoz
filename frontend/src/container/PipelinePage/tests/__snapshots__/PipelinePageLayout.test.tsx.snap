// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelinePage container test should render PipelinePageLayout section 1`] = `
<DocumentFragment>
  .c0.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-bottom: 2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.c1.c1 {
  margin-left: 1rem;
}

<div
    class="c0"
  >
    <span
      aria-label="question-circle"
      class="anticon anticon-question-circle"
      role="img"
      style="font-size: 1rem; color: rgba(255, 255, 255, 0.835);"
    >
      <svg
        aria-hidden="true"
        data-icon="question-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 708c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm62.9-219.5a48.3 48.3 0 00-30.9 44.8V620c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-21.5c0-23.1 6.7-45.9 19.9-64.9 12.9-18.6 30.9-32.8 52.1-40.9 34-13.1 56-41.6 56-72.7 0-44.1-43.1-80-96-80s-96 35.9-96 80v7.6c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V420c0-39.3 17.2-76 48.4-103.3C430.4 290.4 470 276 512 276s81.6 14.5 111.6 40.7C654.8 344 672 380.7 672 420c0 57.8-38.1 109.8-97.1 132.5z"
        />
      </svg>
    </span>
    <button
      class="ant-btn css-dev-only-do-not-override-2i2tap ant-btn-primary c1"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          aria-label="plus"
          class="anticon anticon-plus"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="plus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <defs>
              <style />
            </defs>
            <path
              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
            />
            <path
              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
            />
          </svg>
        </span>
      </span>
      <span>
        new_pipeline
      </span>
    </button>
  </div>
  .c0 {
  margin-top: 3rem;
}

<div
    class="c0"
  >
    <div
      class="logs-pipelines-empty-state-centered-container"
    >
      <div
        class="ant-card ant-card-bordered ant-card-small css-dev-only-do-not-override-2i2tap"
      >
        <div
          class="ant-card-body"
        >
          <div
            class="logs-pipelines-empty-state-centered-container"
          >
            <iframe
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
              allowfullscreen=""
              class="logs-pipelines-empty-state-video-iframe"
              frameborder="0"
              sandbox="allow-scripts allow-same-origin allow-popups allow-popups-to-escape-sandbox"
              src="https://www.youtube.com/embed/OneENGNmLd0"
              title="learn_more"
            />
            <div>
              <article
                class="ant-typography css-dev-only-do-not-override-2i2tap"
              >
                learn_moreÂ 
                <a
                  href="https://signoz.io/docs/logs-pipelines/introduction/?utm_source=product&utm_medium=pipelines-tab"
                  rel="noreferrer"
                  target="_blank"
                >
                  here
                </a>
              </article>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
